array_api_tests/dtype_helpers.py

```diff
@@ -322,15 +322,16 @@ def accumulation_result_dtype(x_dtype, dtype_kwarg):
     default_float = xp.asarray(float()).dtype
     if default_float not in real_float_dtypes:
         warn(f"inferred default float is {default_float!r}, which is not a float")
-    if api_version > "2021.12":
```

Comment:
I don't know about removing this. This is the check that's currently done throughout the suite to check for complex support. 

Maybe we should instead make sure that `ARRAY_API_TESTS_SKIP_DTYPES` works correctly for complex dtypes.

Reply:
I readded the version check and added a check that no complex types are requested to be skipped.